### PR TITLE
[PowerBI] fix issue with special characters in partitioning columns

### PR DESCRIPTION
**Pull request by gbrueckl**
Fixes the issue as described in bug https://github.com/delta-io/delta/issues/2280

#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [x] Other (PowerBI)

## How was this patch tested?
patch was tested with the code provided in the bug description and some additional cases

## Does this PR introduce _any_ user-facing changes?
No

### CODE REVIEW

File: connectors/powerbi/PowerBI_Delta.Dataset/model.bim [hunk 1]
```diff
@@ -349,6 +349,8 @@
           "",
           "        Delimiter = if Text.Contains(DeltaTableFolderContent{0}[Folder Path], \"//\") then \"/\" else \"\\\",",
           "",
+          "        fn_URIDecode = (data as text) as text => Uri.Parts(\"http://powerbi?txt=\" & data)[Query][txt],",
```
Comment (dennyglee) [inline connectors/powerbi/PowerBI_Delta.Dataset/model.bim:352]: Are there any concerns that the Uri.Parts would included `https` not just `http://powerbi`?

Comment (gbrueckl) [reply connectors/powerbi/PowerBI_Delta.Dataset/model.bim:352]: thats just a random name, could also be `https://abc` or `abc://xyz` - we just need something to generate a valid URI to which we pass the encoded string and then we use `Uri.Parts` to extract the all query parameters (`[Query]`) and then select the one we have set to our encoded string (`[txt]`)

Comment (dennyglee) [reply connectors/powerbi/PowerBI_Delta.Dataset/model.bim:352]: Got it!  Thanks!

**Review by dennyglee — APPROVED** [review]
LGTM!

**Pull request merged**